Plugin doesn't seem to work with pipeline-maven-plugin archiver

### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  <!-- Paste your environment details below -->
  ```text
  Paste the output here
  ```

</details>


### What Operating System are you using (both controller, and any agents involved in the problem)?

All

### Reproduction steps

```
withMaven(options: [artifactsPublisher(disabled: false)]) {

}
```

### Expected Results

maven artifact are archived correctly

### Actual Results

```
[10:22:17.036+02:00] - [withMaven] artifactsPublisher - Archive artifact mvn/pom.xml under ****.pom
[10:22:24.174+02:00] - ERROR: [withMaven] WARNING Exception executing Maven reporter 'Generated Artifacts Publisher' / org.jenkinsci.plugins.pipeline.maven.publishers.GeneratedArtifactsPublisher. Please report a bug associated for the component 'pipeline-maven-plugin' at https://issues.jenkins-ci.org/ 
[10:22:24.174+02:00] - java.io.IOException: Exception archiving {****.pom=mvn/pom.xml}
[10:22:24.174+02:00] - 	at org.jenkinsci.plugins.pipeline.maven.publishers.GeneratedArtifactsPublisher.process(GeneratedArtifactsPublisher.java:145)
[10:22:24.174+02:00] - 	at org.jenkinsci.plugins.pipeline.maven.MavenSpyLogProcessor.processMavenSpyLogs(MavenSpyLogProcessor.java:172)
[10:22:24.174+02:00] - 	at org.jenkinsci.plugins.pipeline.maven.WithMavenStepExecution2$WithMavenStepExecutionCallBack.finished(WithMavenStepExecution2.java:1345)
[10:22:24.174+02:00] - 	at org.jenkinsci.plugins.workflow.steps.GeneralNonBlockingStepExecution$TailCall.lambda$onSuccess$0(GeneralNonBlockingStepExecution.java:140)
[10:22:24.174+02:00] - 	at org.jenkinsci.plugins.workflow.steps.GeneralNonBlockingStepExecution.lambda$run$0(GeneralNonBlockingStepExecution.java:77)
[10:22:24.174+02:00] - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
[10:22:24.174+02:00] - 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
[10:22:24.174+02:00] - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[10:22:24.174+02:00] - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[10:22:24.174+02:00] - 	at java.base/java.lang.Thread.run(Unknown Source)
[10:22:24.174+02:00] - Caused by: hudson.remoting.ProxyException: Assertion failed: 
[10:22:24.175+02:00] - 
[10:22:24.175+02:00] - assert content.isFile()
```

### Anything else?

_No response_

### Are you interested in contributing a fix?

Sure

--- a/src/main/java/io/jenkins/plugins/artifactory_artifacts/ArtifactoryArtifactManager.java
+++ b/src/main/java/io/jenkins/plugins/artifactory_artifacts/ArtifactoryArtifactManager.java
@@ -59,7 +59,7 @@ public class ArtifactoryArtifactManager extends ArtifactManager implements Stash
         for (Map.Entry<String, String> entry : artifacts.entrySet()) {
             String path = "artifacts/" + entry.getKey();
             String filePath = getFilePath(path);
-            files.add(new UploadFile(entry.getKey(), filePath));
+            files.add(new UploadFile(entry.getValue(), filePath));
         }
 
         workspace.act(new UploadToArtifactoryStorage(buildArtifactoryConfig(), files));

